src/cirrus/lib2/statedb.py

```diff
@@ -343,7 +350,10 @@ def set_completed(
             UpdateExpression=expr,
             ExpressionAttributeValues=expr_attrs,
         )
-        logger.debug("set outputs", extra=key.update({"outputs": outputs}))
+        logger.debug("set completed", extra=key.update({"outputs": outputs}))
+
+        self.write_timeseries_record(key, StateEnum.COMPLETED, now)
```

Comment:
I feel rather strongly we should make writing the timeseries record conditional on having the execution ARN. If the barrier here is simply passing in that in from `update-state`, I am happy to throw a commit on here to plumb the ARN though.

To recap my concern, we will not be able to group the terminal events for an execution with its start, so we will not be able to calculate things like duration after the fact with certainty that we've related the correct events together.

Another major reason that has solidified my opinion here is using these methods in some ad hoc management scripts I threw together to manage some issues with missed state updates on a recent large ingest. These sorts of things can happen for several reasons when processing millions of items, so being able to mitigate the problems with custom scripts that reuse these methods is super helpful. But in such cases I probably don't want events.